From 52c6799a770a97b82811eb7d104eb3bfd8948c23 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Wed, 22 Jan 2020 20:13:57 +0000
Subject: [PATCH 0989/1077] MANTIS 0057052 Implement service handler status
 request and interface version check

CUSS-1.5.BIOMETRICS.GP.XSD:
add interface-version

CUSS.BIOMETRIC.XSD:
add statusRequest, getInterfaceVersion

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1523 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD | 12 ++++++++++++
 .../Massai/idl/Cuss/CUSS.BIOMETRIC.XSD         | 18 +++++++++++++++++-
 2 files changed, 29 insertions(+), 1 deletion(-)

diff --git a/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD b/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD
index 7c135fef..5ce96efc 100644
--- a/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD
+++ b/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD
@@ -253,4 +253,16 @@
       </xs:enumeration>
     </xs:restriction>
   </xs:simpleType>
+  <xs:simpleType name="interface-version">
+    <xs:annotation>
+      <xs:documentation>Used by service handler implementations, only. Interface version of XB01 protocol implemented by service handler and being checked by CUSS platform's biometrics component. Version must consist of 3 digits (x.y.z). Change last digit (z) for backward-compatible changes only. Other changes require modification of first (x) and/or second (y) digit. For a new interface version, a new enumeration value is required. Existing values must not be modified!</xs:documentation>
+    </xs:annotation>
+    <xs:restriction base="xs:string">
+      <xs:enumeration value="1.0.0">
+        <xs:annotation>
+          <xs:documentation>Very first version</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+    </xs:restriction>
+  </xs:simpleType>
 </xs:schema>
diff --git a/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD b/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD
index 30674858..910c8a0b 100644
--- a/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD
+++ b/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD
@@ -1,5 +1,11 @@
 <xsd:schema xmlns:tns="urn:CUSS-1.5/types/biometrics" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:oneIdTypes="http://www.materna.com/CUSS/OneIdTypes-1.0" targetNamespace="urn:CUSS-1.5/types/biometrics" elementFormDefault="qualified" attributeFormDefault="unqualified">
 
+    <!-- Schema defines the Materna proprietary XB01 (Xml Biometrics Version 01) interface between CUSS platform's biometrics component and the so called 'service handler'.
+        A service handler excaplulates the interface to the customers Biometrics backend (e.g. NEC Face Authentication, Idemia Biopass, CBP TVS, Arinc Self Pass, ...)
+        Current version: 1.0.0 (consider CUSS-1.5.BIOMETRICS-GP.xsd -> 'interface-version' when changing this interface!)
+        Author: sven.bornemann@materna-ips.com 
+    -->
+    
   	<xsd:import namespace="http://www.materna.com/CUSS/OneIdTypes-1.0" schemaLocation="OneIdTypes.xsd"/>
 
     <xsd:simpleType name="score">
@@ -81,6 +87,16 @@
                     <xsd:documentation>Identifies a passenger via Service Handler interface. Pendant for MODI function 'getIdentifiedBiometricImage'. What passenger details are retured depends on the biometics backend</xsd:documentation>
                 </xsd:annotation>
             </xsd:enumeration>
+            <xsd:enumeration value="statusRequest">
+                <xsd:annotation>
+                    <xsd:documentation>Internal command sent by CUSS platform to query service handler's backend connectivity</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
+            <xsd:enumeration value="getInterfaceVersion">
+                <xsd:annotation>
+                    <xsd:documentation>Internal command sent by CUSS platform to query the interface version in order to check the compatiblity between platform and service handler</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
         </xsd:restriction>
     </xsd:simpleType>
 
@@ -438,7 +454,7 @@
             </xsd:element>
             <xsd:element name="message" type="xsd:string" minOccurs="0">
                 <xsd:annotation>
-                    <xsd:documentation>Internal error message or error description from biometrics backend if available</xsd:documentation>
+                    <xsd:documentation>Internal interface version (getInterfaceVersion), status or error message or error description from biometrics backend if available</xsd:documentation>
                 </xsd:annotation>
             </xsd:element>
             <xsd:element name="score" type="tns:score" minOccurs="0">
-- 
2.41.0.windows.1

